src/mpsc/blocking.rs

```diff
@@ -574,6 +575,108 @@ feature! {
+        /// If there are no messages in the channel's buffer, but the channel has
+        /// not yet been closed, this method will block until a message is sent,
+        /// the channel is closed, or the provided `timeout` has elapsed.
+        ///
+        /// # Returns
+        ///
+        /// - [`Ok`]`(`[`RecvRef`]`<T>)` if a message was received.
+        /// - [`Err`]`(`[`RecvTimeoutError::Timeout`]`)` if the timeout has elapsed.
+        /// - [`Err`]`(`[`RecvTimeoutError::Closed`]`)` if the channel has closed.
+        /// # Examples
+        ///
+        /// ```
+        /// use thingbuf::mpsc::{blocking::StaticChannel, errors::RecvTimeoutError};
+        /// use std::{thread, fmt::Write, time::Duration};
+        ///
+        /// static CHANNEL: StaticChannel<String, 100> = StaticChannel::new();
+        /// let (tx, rx) = CHANNEL.split();
+        ///
+        /// thread::spawn(move || {
+        ///     thread::sleep(Duration::from_millis(150));
+        ///     let mut value = tx.send_ref().unwrap();
+        ///     write!(value, "hello world!")
+        ///         .expect("writing to a `String` should never fail");
+        /// });
+        ///
+        /// assert_eq!(
+        ///     Err(&RecvTimeoutError::Timeout),
+        ///     rx.recv_ref_timeout(Duration::from_millis(100)).as_deref().map(String::as_str)
+        /// );
+        /// assert_eq!(
+        ///     Ok("hello world!"),
+        ///     rx.recv_ref_timeout(Duration::from_millis(100)).as_deref().map(String::as_str)
+        /// );
+        /// assert_eq!(
+        ///     Err(&RecvTimeoutError::Closed),
+        ///     rx.recv_ref_timeout(Duration::from_millis(100)).as_deref().map(String::as_str)
+        /// );
+        /// ```
+        pub fn recv_ref_timeout(&self, timeout: Duration) -> Result<RecvRef<'_, T>, RecvTimeoutError> {
+            recv_ref_timeout(self.core, self.slots, timeout)
+        }
+
+        /// Receives the next message for this receiver, **by value**, waiting for at most `timeout`.
+        ///
+        /// If there are no messages in the channel's buffer, but the channel
+        /// has not yet been closed, this method will block until a message is
+        /// sent, the channel is closed, or the provided `timeout` has elapsed.
+        ///
+        /// When a message is received, it is moved out of the channel by value,
+        /// and replaced with a new slot according to the configured [recycling
+        /// policy]. If all [`StaticSender`]s for this channel write to the
+        /// channel's slots in place by using the [`send_ref`] or
+        /// [`try_send_ref`] methods, consider using the [`recv_timeout_ref`]
+        /// method instead, to enable the reuse of heap allocations.
+        ///
+        /// [`send_ref`]: StaticSender::send_ref
+        /// [`try_send_ref`]: StaticSender::try_send_ref
+        /// [recycling policy]: crate::recycling::Recycle
+        /// [`recv_timeout_ref`]: Self::recv_timeout_ref
```

Comment:
whoops, my bad, i got the wrong method name here and broke CI:
```suggestion
        /// [`try_send_ref`] methods, consider using the [`recv_ref_timeout`]
        /// method instead, to enable the reuse of heap allocations.
        ///
        /// [`send_ref`]: StaticSender::send_ref
        /// [`try_send_ref`]: StaticSender::try_send_ref
        /// [recycling policy]: crate::recycling::Recycle
        /// [`recv_ref_timeout`]: Self::recv_ref_timeout
```